src/PlatformEngines.jl

```diff
@@ -16,11 +16,22 @@ const exe7z = Ref("")
 function __init__()
     exe7z[] = ""
     name = "7z"
-    Sys.iswindows() && (name = "$name.exe")
-    for dir in (joinpath("..", "libexec"), ".")
-        path = normpath(Sys.BINDIR, dir, name)
-        isfile(path) || continue
-        exe7z[] = path
+    if Sys.iswindows()
+        name = "$name.exe"
+    end
+
+    paths = [
+        joinpath(Sys.BINDIR, Base.LIBEXECDIR),
+        Sys.BINDIR,
+    ]
+    PATH = get(ENV, "PATH", nothing)
+    if PATH !== nothing
+        push!(paths, PATH)
```

Comment:
Why are you pushing the value of `PATH` onto `paths` when it's a list of directories to look in?